docs/rfc/rfc-005-accounts-auth.md

```diff
@@ -0,0 +1,299 @@
+  string address = 1;
+  uint64 id = 2;
+  google.Protobuf.Any data = 3;
+}
+
+message MsgExecute {
+  string sender = 1;
+  string address = 2;
+  google.Protobuf.Any message = 3;
+  repeated cosmos.base.v1beta1.Coin funds = 4 [(gogoproto.nullable) = false];
+}
+
+message MsgExecuteResponse {
+  google.Protobuf.Any data = 1;
+}
+```
+
+#### MsgDeploy
+
+Deploys a new instance of the given account `kind` with initial settings represented by the `init_message` which is a `google.Protobuf.Any`.
+Of course the `init_message` can be empty. A response is returned containing the account ID and humanised address, alongside some response
+that the account instantiation might produce.
+
+#### MsgExecute
+
+Sends a `StateTransition` execution request, where the state transition is represented by the `message` which is a `google.Protobuf.Any`.
+The account can then decide if to process it or not based on the `sender`.
+
+### Further discussion
+
+#### Sub-accounts
+
+We could provide a way to link accounts to other accounts. Maybe during deployment the sender could decide to link the 
+newly created to its own account, although there might be use-cases for which the deployer is different from the account
+that needs to be linked, in this case a handshake protocol on linking would need to be defined.
+
+#### Predictable address creation
+
+We need to provide a way to create an account with a predictable address, this might serve a lot of purposes, like accounts
+wanting to generate an address that:
+- nobody else can claim besides the account used to generate the new account
+- is predictable
+
+For example:
+```protobuf
+
+message MsgDeployPredictable {
+  string sender = 1;
+  uint32 nonce = 2; 
+  ...
+}
+```
+
+And then the address becomes `bechify(concat(sender, nonce))`
+
+`x/accounts` would still use the monotonically increasing sequence as account number.
+
+## x/authn and account credentials
+
+We have initially mentioned that `x/accounts` only provides for a way to deploy accounts, but they're not backed by any 
+credential, which means no state transition on behalf of the account can be incepted from a TX.
+
+`x/authn` is meant to provide the link between accounts and their TX credentials.
+Specifically `x/authn` defines the following `tx` gRPC interface:
+
+```protobuf
+// Msg defines the Msg service.
+service Msg {
+  rpc CreateAuthenticatedAccount(MsgCreateAuthenticatedAccount) returns (MsgCreateAuthenticatedAccountResponse) {};
+  rpc UpdateCredentials(MsgUpdateCredentials) returns (MsgUpdateCredentialsResponse) {};
+  rpc DeleteCredentials(MsgDeleteCredentials) returns (MsgDeleteCredentials) {};
+}
+
+message MsgCreateAuthenticatedAccount {
+  string sender = 1;
+  google.Protobuf.Any credential = 2;
+  accounts.MsgDeploy deploy_msg = 3;
+}
+
+message MsgCreateAuthenticatedAccountResponse {
+  accounts.MsgDeployResponse deploy_response = 1;
+}
+
+message MsgUpdateCredentials {
+  string sender = 1;
+  string kind = 2;
```

Comment:
Are these `kind` fields needed? How are they different from they `type_url` in `Any`?